Fix MOM6SOLO build in the EMC branch.

There is an issue with compiling the standalone MOM6 within UFS (or even without UFS) in the [dev/emc](https://github.com/NOAA-EMC/MOM6/tree/dev/emc) branch

There was a bug in the way the wave forcing subroutine was being called with incorrect number of arguments.

To reproduce the compilation error from the UFS-weather-model repo

```
CMAKE_FLAGS="-DMOM6SOLO=ON -DS2S=ON -DDATM=ON" BUILD_VERBOSE=1 BUILD_JOBS=1 BUILD_DIR=build-datmmom6 ./build.sh
```

```
[ 99%] Building Fortran object MOM6-interface/CMakeFiles/mom6solo.dir/MOM6/config_src/solo_driver/MOM_driver.F90.o
cd /Users/rmahajan/scratch/UFSwork/weather-develop/build-datmmom6/MOM6-interface && /opt/modules/gnu-9.3.0/mpich/3.3.1/bin/mpifort -DENABLE_QUAD_PRECISION -DGFS_PHYS -DINTERNAL_FILE_NML -Duse_libMPI -Duse_netCDF -I/Users/rmahajan/scratch/UFSwork/weather-develop/build-datmmom6/MOM6-interface/mod -I/Users/rmahajan/scratch/UFSwork/weather-develop/build-datmmom6/MOM6-interface/mod_solo -I/Users/rmahajan/scratch/UFSwork/weather-develop/MOM6-interface/MOM6/config_src/dynamic -I/Users/rmahajan/scratch/UFSwork/weather-develop/MOM6-interface/MOM6/src/framework -I/Users/rmahajan/scratch/UFSwork/weather-develop/FMS/include -I/Users/rmahajan/scratch/UFSwork/weather-develop/FMS/fms -I/Users/rmahajan/scratch/UFSwork/weather-develop/FMS/mpp/include -I/Users/rmahajan/scratch/UFSwork/weather-develop/build-datmmom6/FMS/mod -I/opt/modules/gnu-9.3.0/mpich-3.3.1/netcdf/4.7.4/include -g -fbacktrace -fdefault-real-8 -fdefault-double-8 -Waliasing -fcray-pointer -fconvert=big-endian -ffree-line-length-none -fno-range-check -fbacktrace -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.0.sdk -mmacosx-version-min=10.15 -Jmod_solo -c /Users/rmahajan/scratch/UFSwork/weather-develop/MOM6-interface/MOM6/config_src/solo_driver/MOM_driver.F90 -o CMakeFiles/mom6solo.dir/MOM6/config_src/solo_driver/MOM_driver.F90.o
/Users/rmahajan/scratch/UFSwork/weather-develop/MOM6-interface/MOM6/config_src/solo_driver/MOM_driver.F90:491:79:

  491 |       call Update_Surface_Waves(grid, GV, US, time, time_step_ocean, waves_csp)
      |                                                                               1
Error: Missing actual argument for argument 'forces' at (1)
make[2]: *** [MOM6-interface/CMakeFiles/mom6solo.dir/MOM6/config_src/solo_driver/MOM_driver.F90.o] Error 1
make[1]: *** [MOM6-interface/CMakeFiles/mom6solo.dir/all] Error 2
make: *** [all] Error 2
```

--- a/src/user/MOM_wave_interface.F90
+++ b/src/user/MOM_wave_interface.F90
@@ -452,7 +452,7 @@ subroutine Update_Surface_Waves(G, GV, US, Day, dt, CS, forces)
   type(unit_scale_type),   intent(in)  :: US   !< A dimensional unit scaling type
   type(time_type),         intent(in)  :: Day !< Current model time
   type(time_type),         intent(in)  :: dt  !< Timestep as a time-type
-  type(mech_forcing),      intent(in)  :: forces !< MOM_forcing_type
+  type(mech_forcing),      intent(in), optional  :: forces !< MOM_forcing_type
   ! Local variables
   integer :: ii, jj, kk, b
   type(time_type) :: Day_Center

